src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -25,8 +25,13 @@ public string Serialize(object sourceObject)
             {
                 formatter.Serialize(outputStream, sourceObject);
 
+#if DOTNET5_4
+                ArraySegment<byte> buffer;
+                var gotBuffer = outputStream.TryGetBuffer(out buffer);
+                var outputBytes = gotBuffer ? buffer.Array : new byte[] { };

```

Comment:
This isn't right. You need to use the array segment length and offset. Better to always use an array segment:

```C#
var buffer = default(ArraySegment<byte>);

#if DOTNET5_4
    outputStream.TryGetBuffer(out buffer);
#else
    var outputBytes = outputStream.GetBuffer();
    buffer = new ArraySegment<byte>(outputBytes, 0, outputBytes.Length);
#endif
    return Convert.ToBase64String(buffer.Array, buffer.Offset, buffer.Count);
```

Something like that